src/TorchSharp/Optimizers/SGD.cs

```diff
@@ -233,6 +233,7 @@ public override void to(Device device)
                 public override void LoadStateDict(BinaryReader reader)
                 {
                     LoadConditionalStateTensor(reader, ref momentum_buffer);
+                    momentum_buffer = momentum_buffer.to(_parameter.device); 
```

Comment:
No need to dispose the old momentum buffer if it's actually moved?

Reply:
Good catch. Will fix.

Reply:
Question: Do you want me to write something like:
```cs
if (momentum_buffer.device_type != _parameter.device_type || momentum_buffer.device_index != _parameter.device_index) {
    using var copy = result;
    result = copy.to(_parameter.device);
}
```

Reply:
(That's not handled in the regular `Optimizer.to(Device device)` function. See [here](https://github.com/dotnet/TorchSharp/blob/main/src/TorchSharp/Optimizers/SGD.cs#L226))

Reply:
I think it's sufficient to check the `Handle` before and after the call to `to'. I think.

Reply:
It's handled in the Module variants of `to()`. Follow that pattern.

Reply:
Maybe it's worth writing an internal utility function that has the 'move or not move' predicate, so it can be reused.